Test(UI): Add E2E tests for Asset Details

junaiddshaukat opened the pull request:
This adds Page Object and specs for verifying the Asset Details page, covering producing tasks and consuming DAGs validation.

<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

This PR adds Playwright UI E2E coverage for the Asset Details page.
Adds a page object (AssetsPage) for navigating Assets and asserting details/dependencies.
Adds an E2E spec verifying an asset’s details include producing tasks and scheduled DAGs.
closes: #59930

<!-- Please keep an empty line above the dashes. -->
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [airflow-core/newsfragments](https://github.com/apache/airflow/tree/main/airflow-core/newsfragments).

junaiddshaukat committed:
Test(UI): Add E2E tests for Asset Details

This adds Page Object and specs for verifying the Asset Details page,
covering producing tasks and consuming DAGs validation.

Signed-off-by: junaiddshaukat <junaiddshaukat@gmail.com>

vatsrahul1001 commented on airflow-core/src/airflow/ui/tests/e2e/specs/assets.spec.ts:
```diff
@@ -0,0 +1,41 @@
+
+test.describe("Assets Page", () => {
+  let assetsPage: AssetsPage;
+  const assetName = "s3://dag1/output_1.txt";
```
This assumes a specific asset exists. If CI environment doesn't have this asset, tests will fail.

Add beforeAll to trigger the producer DAG

```
test.beforeAll(async ({ browser }) => {
  const context = await browser.newContext({ storageState: AUTH_FILE });
  const page = await context.newPage();
  const dagsPage = new DagsPage(page);
  
  // Trigger asset_produces_1 to create asset data
  await dagsPage.triggerDag("asset_produces_1");
  // Wait for completion...
  
  await context.close();
});
```

vatsrahul1001 replied:
Can we get this from config

vatsrahul1001 commented on airflow-core/src/airflow/ui/tests/e2e/pages/AssetsPage.ts:
```diff
@@ -0,0 +1,98 @@
+
+    // Strategy: Get the parent of the label, then find the button inside it.
+    // Assuming Stat component structure.
+    const statBox = label.locator("..");
```
This relies on specific DOM structure. If UI changes, this breaks.

vatsrahul1001 commented on airflow-core/src/airflow/ui/tests/e2e/pages/AssetsPage.ts:
```diff
@@ -0,0 +1,98 @@
+    if (count > 0) {
+      await button.click();
+      // Verify popover content (links)
+      await expect(this.page.locator(".chakra-popover__body a")).toHaveCount(count);
```
Framework-specific class names can change

vatsrahul1001 commented on airflow-core/src/airflow/ui/tests/e2e/pages/AssetDetailPage.ts:
```diff
@@ -0,0 +1,98 @@
+    await expect(this.page.getByRole("heading", { name })).toBeVisible();
+  }
+
+  public async verifyProducingTasks(minCount: number): Promise<void> {
```
verifyProducingTasks and verifyScheduledDags are nearly identical. We should refactor

junaiddshaukat committed:
fix: address PR review feedback for asset details E2E tests

- Rename AssetsPage to AssetDetailPage for clarity (coordinates with #59990)
- Add beforeAll to trigger asset_produces_1 DAG before tests
- Fix fragile selectors (use parent container + ARIA roles)
- Refactor duplicate code into verifyStatSection helper
- Remove unused assetsTable locator
- Rename test to remove 'should' prefix
- Change describe name to 'Asset Details Page'

junaiddshaukat committed:
Merge branch 'main' into test/ui-asset-details

vatsrahul1001 committed:
Merge branch 'main' into test/ui-asset-details

vatsrahul1001 commented on airflow-core/src/airflow/ui/tests/e2e/specs/assets.spec.ts:
```diff
@@ -0,0 +1,73 @@
+/*!
```
Add the test to the existing asset.spec.ts

vatsrahul1001 commented on airflow-core/src/airflow/ui/tests/e2e/specs/assets.spec.ts:
```diff
@@ -0,0 +1,73 @@
+  let assetDetailPage: AssetDetailPage;
+  const assetName = "s3://dag1/output_1.txt";
+
+  test.beforeAll(async ({ browser }) => {
```
asset.spec.ts already have beforeALL if you add this test to same test spec we can use same

junaiddshaukat committed:
Merge branch 'main' into test/ui-asset-details

junaiddshaukat committed:
refactor: merge asset detail test into existing asset.spec.ts

- Add asset.name to testConfig in playwright.config.ts
- Merge asset detail test into existing asset.spec.ts (reuse beforeAll)
- Use testConfig.asset.name instead of hardcoded values
- Remove duplicate assets.spec.ts file

vatsrahul1001 committed:
Merge branch 'main' into test/ui-asset-details
